nre/bpf/nodefw/nodefw/src/drainer.rs

```diff
@@ -0,0 +1,23 @@
+use std::fs;
+use tokio_util::sync::CancellationToken;
+use tracing::info;
+
+pub async fn watch(ctx: CancellationToken, path: &str) {
+    let mut interval = tokio::time::interval(std::time::Duration::from_secs(1));
+    interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+
+    loop {
+        tokio::select! {
+            _ = interval.tick() => {
+                if let Ok(_) = fs::metadata(path) {
```

Comment:
OOC, why not path.exists() ?

Reply:
in the event i want access to the metadata info for some reason.

https://doc.rust-lang.org/std/fs/struct.Metadata.html